src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -346,6 +334,57 @@ public virtual VerifyResult Verify(StoreView snapshot, BigInteger totalSenderFee
             return VerifyResult.Succeed;
         }
 
+        #region Oracles
+
+        private void LoadOracleAttribute()
+        {
+            if (oracleCached != OracleCache.None)
+            {
+                return;
+            }
+
+            var attr = attributes.Where(u => u.Usage == TransactionAttributeUsage.OracleRequest).FirstOrDefault();
+
+            if (attr != null)
+            {
+                oracleCached |= OracleCache.Request;
+            }
+
+            attr = attributes.Where(u => u.Usage == TransactionAttributeUsage.OracleResponse).FirstOrDefault();
+
+            if (attr != null)
+            {
+                oracleCached |= OracleCache.Response;
+                oracleResponse = attr.Data.AsSerializable<OracleResponseAttribute>();
+            }
+            else
+            {
+                oracleResponse = null;
+            }
```

Comment:
If you need to check the cosigners, you need to fetch all the attributes and find/parse the expected type. Is the same logic that I need for check if it's a oracle request, or oracle response or both.

Reply:
@erikzhang please take a look to https://github.com/neo-project/neo/pull/1620